core/services/wifi/WifiManager.py

```diff
@@ -81,13 +86,38 @@ def __dict_from_list(data: bytes) -> Dict[str, Any]:
 
     async def get_wifi_available(self) -> List[ScannedWifiNetwork]:
         """Get a dict from the wifi signals available"""
-        try:
-            await self.wpa.send_command_scan(timeout=15)
-            data = await self.wpa.send_command_scan_results()
-            networks_list = WifiManager.__dict_from_table(data)
-            return [ScannedWifiNetwork(**network) for network in networks_list]
-        except Exception as error:
-            raise FetchError("Failed to fetch wifi list.") from error
+
+        async def perform_new_scan() -> None:
+            try:
+                # We store the scan task here so that new scan requests that happen in the interval
+                # where this one is running can check when it has finished.
+                # Otherwise, it could happen that a new scan is initiated before the ones waiting know
+                # the previous one has finished, making them stay on the loop unnecessarily.
+                self._scan_task = asyncio.create_task(self.wpa.send_command_scan(timeout=15))
+                await self._scan_task
+                data = await self.wpa.send_command_scan_results()
+                networks_list = WifiManager.__dict_from_table(data)
+                self._updated_scan_results = [ScannedWifiNetwork(**network) for network in networks_list]
+            except Exception as error:
+                if self._scan_task is not None:
+                    self._scan_task.cancel()
+                self._updated_scan_results = None
+                raise FetchError("Failed to fetch wifi list.") from error
+
+        # Performs a new scan only if it's the first one or the last one is already done
+        # In case there's one running already, wait for it to finish and use its result
+        if self._scan_task is None or self._scan_task.done():
+            await perform_new_scan()
+        else:
+            awaited_scan_task_name = self._scan_task.get_name()
+            while self._scan_task.get_name() == awaited_scan_task_name and not self._scan_task.done():
```

Comment:
The comment before tries to explain the problem and why do we use this.
Think about the following scenario:
1. API receives scan call.
2. No scan operation is in place, so a new one is started
3. API receives another scan call.
4. There's already a scan operation in place, so it sleeps, waiting for it to be finished
5. First scan is finished and gets out of the loop
6. API receives another scan call.
7. No scan operation is in place, so a new one is started
8. Second scan call gets out of the sleep and checks if the scan task is done, but it is not (the third call restarted it), so it sleeps again

Got how the second scan got trapped by the third one? With the name approach, we guarantee this won't happen, since it checks if the scan that is in place is the one it was waiting the first time.